--- a/src/http/ngx_http_script.c
+++ b/src/http/ngx_http_script.c
@@ -688,21 +688,22 @@
 
     zero = (sc->zero && last);
     len = value->len + zero;
 
     code = ngx_http_script_add_code(*sc->lengths,
                                     sizeof(ngx_http_script_copy_code_t), NULL);
     if (code == NULL) {
         return NGX_ERROR;
     }
 
-    code->code = (ngx_http_script_code_pt) ngx_http_script_copy_len_code;
+    code->code = (ngx_http_script_code_pt) (void *)
+                                                 ngx_http_script_copy_len_code;
     code->len = len;
 
     size = (sizeof(ngx_http_script_copy_code_t) + len + sizeof(uintptr_t) - 1)
             & ~(sizeof(uintptr_t) - 1);
 
     code = ngx_http_script_add_code(*sc->values, size, &sc->main);
     if (code == NULL) {
         return NGX_ERROR;
     }
 
@@ -777,21 +778,22 @@
 
         *p = index;
     }
 
     code = ngx_http_script_add_code(*sc->lengths,
                                     sizeof(ngx_http_script_var_code_t), NULL);
     if (code == NULL) {
         return NGX_ERROR;
     }
 
-    code->code = (ngx_http_script_code_pt) ngx_http_script_copy_var_len_code;
+    code->code = (ngx_http_script_code_pt) (void *)
+                                             ngx_http_script_copy_var_len_code;
     code->index = (uintptr_t) index;
 
     code = ngx_http_script_add_code(*sc->values,
                                     sizeof(ngx_http_script_var_code_t),
                                     &sc->main);
     if (code == NULL) {
         return NGX_ERROR;
     }
 
     code->code = ngx_http_script_copy_var_code;
@@ -1171,22 +1173,22 @@
 {
     ngx_http_script_copy_capture_code_t  *code;
 
     code = ngx_http_script_add_code(*sc->lengths,
                                     sizeof(ngx_http_script_copy_capture_code_t),
                                     NULL);
     if (code == NULL) {
         return NGX_ERROR;
     }
 
-    code->code = (ngx_http_script_code_pt)
-                      ngx_http_script_copy_capture_len_code;
+    code->code = (ngx_http_script_code_pt) (void *)
+                                         ngx_http_script_copy_capture_len_code;
     code->n = 2 * n;
 
 
     code = ngx_http_script_add_code(*sc->values,
                                     sizeof(ngx_http_script_copy_capture_code_t),
                                     &sc->main);
     if (code == NULL) {
         return NGX_ERROR;
     }
 
@@ -1286,21 +1288,22 @@
 {
     ngx_http_script_full_name_code_t  *code;
 
     code = ngx_http_script_add_code(*sc->lengths,
                                     sizeof(ngx_http_script_full_name_code_t),
                                     NULL);
     if (code == NULL) {
         return NGX_ERROR;
     }
 
-    code->code = (ngx_http_script_code_pt) ngx_http_script_full_name_len_code;
+    code->code = (ngx_http_script_code_pt) (void *)
+                                            ngx_http_script_full_name_len_code;
     code->conf_prefix = sc->conf_prefix;
 
     code = ngx_http_script_add_code(*sc->values,
                                     sizeof(ngx_http_script_full_name_code_t),
                                     &sc->main);
     if (code == NULL) {
         return NGX_ERROR;
     }
 
     code->code = ngx_http_script_full_name_code;
